test/addons/openssl-client-cert-engine/binding.gyp

```diff
@@ -0,0 +1,24 @@
+{
+  'includes': ['../../../config.gypi'],
+
+  'targets': [
+    {
+      'target_name': 'testengine',
+      'conditions': [
+        ['node_use_openssl=="true"', {
+          'type': 'shared_library',
+          'sources': [ 'testengine.cc' ],
+          'product_extension': 'engine'
+        }],
+        ['node_use_openssl=="true" and node_shared_openssl=="false"', {
+          'include_dirs': ['../../../deps/openssl/openssl/include'],
+          'link_settings': {
+            'libraries': [
+              '<(PRODUCT_DIR)/../../../../../out/<(PRODUCT_DIR)/libopenssl.a',
```

Comment:
If this is purely for testing then you can do something ugly like:
```python
             'dependencies': [
                '../../../deps/openssl/openssl.gyp:openssl'
              ]
```
Instead of the whole `link_settings` field

Reply:
Maybe on to something, but locally for me, that seems to add a *lot* of compile time and then fails with:

```console
Path: addons/openssl-client-cert-engine/test
_tls_common.js:201
      c.context.setClientCertEngine(options.clientCertEngine);
                ^

Error: error:25066067:DSO support routines:DLFCN_LOAD:could not load the shared library
    at Object.createSecureContext (_tls_common.js:201:17)
    at Object.exports.connect (_tls_wrap.js:1053:48)
    at Agent.createConnection (https.js:111:22)
    at Agent.createSocket (_http_agent.js:217:26)
    at Agent.addRequest (_http_agent.js:187:10)
    at new ClientRequest (_http_client.js:271:16)
    at Object.request (http.js:39:10)
    at Object.request (https.js:243:15)
    at Server.https.createServer.listen (/Users/trott/io.js/test/addons/openssl-client-cert-engine/test.js:46:21)
    at Object.onceWrapper (events.js:314:30)
Command: out/Release/node /Users/trott/io.js/test/addons/openssl-client-cert-engine/test.js
```

Reply:
Oh I think maybe it's just missing a `..`? Let me try again...

Reply:
No, that wasn't it. It does compile stuff in `test/deps` though so I imagine some setting or path is wrong...